typedb/connection/database.py

```diff
@@ -19,25 +19,95 @@
 # under the License.
 #
 
-from typedb.api.connection.database import Database
-from typedb.common.rpc.request_builder import core_database_schema_req, core_database_delete_req
-from typedb.common.rpc.stub import TypeDBStub
+from __future__ import annotations
+from typing import Optional, TYPE_CHECKING
 
+from typedb.api.connection import database
+from typedb.common.exception import TypeDBClientException, DATABASE_DELETED
+from typedb.common.streamer import Streamer
+from typedb.typedb_client_python import database_get_name, database_schema, database_delete, database_rule_schema, \
+    database_type_schema, ReplicaInfo, replica_info_get_address, replica_info_is_primary, replica_info_is_preferred, \
+    replica_info_get_term, database_get_replicas_info, database_get_primary_replica_info, \
+    database_get_preferred_replica_info, replica_info_iterator_next
 
-class _TypeDBDatabaseImpl(Database):
+if TYPE_CHECKING:
+    from typedb.typedb_client_python import Database as NativeDatabase
 
-    def __init__(self, stub: TypeDBStub, name: str):
-        self._name = name
-        self._stub = stub
+
+class _Database(database.Database):
+
+    def __init__(self, database: NativeDatabase):
+        self._native_object = database
+        self._name = database_get_name(database)
+
+    @property
+    def native_object(self):
+        return self._native_object
 
     def name(self) -> str:
+        if not self.native_object.thisown:
```

Comment:
What is `.thisown`?

Reply:
`.thisown` indicates who owns the object. If it is set to 1, Python is the owner, otherwise the Rust side is.